lib/listen/event/queue.rb

```diff
@@ -36,17 +35,15 @@ def <<(args)
         Listen::Logger.debug("InvocaDebug: queuing #{[type, change, dir, path, options]} to queue with depth #{event_queue.size}")
 
         event_queue.public_send(:<<, [type, change, dir, path, options])
-
-        block.call(args) if block
```

Comment:
This whole block thing was ridiculous--they were using it to wakeup the thread that was blocked in `sleep`. Whereas now the thread is simply blocked in `Queue#pop` where it will automatically wake up.